src/util/result.h

```diff
@@ -21,9 +21,9 @@ struct Error {
 //!
 //! It is intended for high-level functions that need to report error strings to
 //! end users. Lower-level functions that don't need this error-reporting and
-//! only need error-handling should avoid util::Result and instead use standard
-//! classes like std::optional, std::variant, and std::tuple, or custom structs
-//! and enum types to return function results.
+//! that only need error-handling should avoid util::Result and instead use
+//! util::Expected, or custom structs and enum types to return function
```

Comment:
In commit "Add util::Expected (std::expected)" (fa1eee4f46fd089d6e19b124addee7c7679585f4)

IMO std::optional and std::variant are still appropriate to use in many cases and it would nice to keep mentioning them.